src/components/Advertisement/index.js

```diff
@@ -15,55 +17,56 @@ const Advertisement = ({
-        <Box display={'flex'} sx={styles.root} gap={20}>
-          <Box
-            component="img"
-            sx={{
-              height: 140,
-              width: 180,
-              objectFit: 'cover',
-              borderRadius: 25,
-            }}
-            src={image}
-          />
-          <Box
-            display={'flex'}
-            flexGrow={10}
-            flexDirection={'column'}
-            justifyContent={'space-between'}
-          >
-            <Box display={'flex'} flexDirection={'column'}>
-              {renderTags(activities, 'activitiesTags', 2)}
-              {renderTags(animals, 'animalsTags', 2)}
-            </Box>
-            <Box display={'flex'} alignItems={'center'}>
-              <Icon Component={FmdGoodIcon} size={'large'}></Icon>
-              <Typography variant={'h1'}>{location}</Typography>
-            </Box>
-          </Box>
-          <Box
-            display={'flex'}
-            flexGrow={4}
-            flexDirection={'column'}
-            justifyContent={'space-between'}
-          >
-            <Box display={'flex'} justifyContent={'flex-end'}>
-              <Rating value={starsValue} />
+    <Paper sx={styles.elevation}>
+      <Collapse in={isExpanded} collapsedSize={170} sx={styles.collapse}>
+        <Box sx={styles.root}>
+          <Box onClick={() => setExpanded(!isExpanded)} sx={styles.rootBox}>
+            <Box sx={styles.collapsedBox}>
+              <Box component="img" sx={styles.imageBox} src={image} />
+              <Box sx={styles.centerColumnBox}>
+                <Box sx={styles.tagsBox}>
+                  {renderTags(activities, 'activitiesTags', 2)}
+                  {renderTags(animals, 'animalsTags', 2)}
+                </Box>
+                <Box sx={styles.locationBox}>
+                  <Icon Component={FmdGoodIcon} size={'large'}></Icon>
+                  <Typography variant={'h1'}>{location}</Typography>
+                </Box>
+              </Box>
+              <Box sx={styles.rightColumnBox}>
+                <Box sx={styles.justifyEndBox}>
+                  <Rating value={starsValue} />
+                </Box>
+                <Box sx={styles.justifyEndBox}>
+                  <PriceRange
+                    from={price.from}
+                    to={price.to}
+                    type={price.priceType}
+                  />
+                </Box>
+              </Box>
             </Box>
-            <Box display={'flex'} justifyContent={'flex-end'}>
-              <PriceRange
-                from={price.from}
-                to={price.to}
-                type={price.priceType}
-              />
+            <Box sx={styles.expandedBox}>
+              <Typography variant="h2">{description}</Typography>
+              <Box sx={styles.availabilityBox}>Availability</Box>
+              <Box sx={styles.justifyEndBox}>
+                <Box sx={styles.justifyEndBox}>
+                  <Button type="submit" sx={styles.contactButton}>
```

Comment:
type="submit" powoduje to, że jak mamy formularz (czyli u nas za pomocą tagów Formik i Form) to po kliknięciu buttona automatycznie wywoła się funkcja onSubmit przekazana do Formika. Tutaj nie masz Formika (i nie potrzebujemy go bo nie ma żadnych inputów itp) więc można wywalić ten type="submit" a kiedyś dołoży się onClick z jakimś handlerem.